controllers/perconaserverformysql_controller.go

```diff
@@ -165,6 +172,213 @@ func (r *PerconaServerForMySQLReconciler) ensureUserSecrets(
 	return nil
 }
 
+func (r *PerconaServerForMySQLReconciler) reconcileUsers(ctx context.Context, cr *apiv2.PerconaServerForMySQL) error {
+	l := log.FromContext(ctx).WithName("reconcileUsers")
+
+	if cr.Status.MySQL.State != apiv2.StateReady {
+		l.Info("MySQL is not ready")
+		return nil
+	}
+
+	secret := &corev1.Secret{}
+	nn := types.NamespacedName{Name: cr.Spec.SecretsName, Namespace: cr.Namespace}
+	if err := r.Client.Get(ctx, nn, secret); err != nil {
+		return errors.Wrapf(err, "get Secret/%s", nn.Name)
+	}
+
+	internalSecret := &corev1.Secret{}
+	nn.Name = cr.InternalSecretName()
+	err := r.Client.Get(ctx, nn, internalSecret)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrapf(err, "get Secret/%s", nn.Name)
+	}
+
+	// Internal secret is not found
+	if k8serrors.IsNotFound(err) {
+		secret.DeepCopyInto(internalSecret)
+		internalSecret.ObjectMeta = metav1.ObjectMeta{
+			Name:      cr.InternalSecretName(),
+			Namespace: cr.Namespace,
+		}
+
+		if err := r.Client.Create(ctx, internalSecret); err != nil {
+			return errors.Wrapf(err, "create Secret/%s", internalSecret.Name)
+		}
+
+		return nil
+	}
+
+	hash, err := k8s.ObjectHash(secret)
+	if err != nil {
+		return errors.Wrapf(err, "get secret/%s hash", secret.Name)
+	}
+
+	internalHash, err := k8s.ObjectHash(internalSecret)
+	if err != nil {
+		return errors.Wrapf(err, "get secret/%s hash", internalSecret.Name)
+	}
+
+	if hash == internalHash {
+		l.V(1).Info("Secret data is up to date")
+		return nil
+	}
+
+	var (
+		restartMySQL        bool
+		restartReplication  bool
+		restartOrchestrator bool
+	)
+	updatedUsers := make([]mysql.User, 0)
+	for user, pass := range secret.Data {
+		if bytes.Equal(pass, internalSecret.Data[user]) {
+			l.V(1).Info("User password is up to date", "user", user)
+			continue
+		}
+
+		mysqlUser := mysql.User{
+			Username: apiv2.SystemUser(user),
+			Password: string(pass),
+			Hosts:    []string{"%"},
```

Comment:
I belive `UserClusterCheck`, `UserMonitor`, `UserXtraBackup` users needs `localhost` only priv.
please fix also `build/ps-entrypoint.sh`

Reply:
`UserXtraBackup` has `%` in PXC: https://github.com/percona/percona-xtradb-cluster-operator/blob/main/build/pxc-entrypoint.sh#L373-L374
`UserMonitor` also has `%` in PXC: https://github.com/percona/percona-xtradb-cluster-operator/blob/main/pkg/pxc/app/statefulset/node.go#L148-L155
`UserClusterCheck`: https://github.com/percona/percona-server-mysql-operator/pull/26/commits/0d0e0eff69097a38a877b5db94887ff8234ae21a

Reply:
still needed to fix permissions for `UserXtraBackup` and add `SYSTEM_USER ` permission to all users except `root`

Reply:
@egegunes how to add `SYSTEM_USER` permission you can find in this commit 
https://github.com/percona/percona-xtradb-cluster-operator/commit/97ce6ed93608f9d07333bd78551b266c51351a00#diff-a54fdab99545e739aea21a652f2e230edaaa7d44aaa13e83a1adda24803db444

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/26/commits/e289374c4f2268d1097fdf27f49a289c8e0fd94d